src/drawer.js

```diff
@@ -48,18 +48,17 @@ WaveSurfer.Drawer = {
         var nominalWidth = this.width;
         var parentWidth = this.getWidth();
 
-        var progress;
-
         if (!this.params.fillParent && nominalWidth < parentWidth) {
-            progress = ((clientX - bbox.left) * this.params.pixelRatio / nominalWidth) || 0;
-
-            if (progress > 1) {
-                progress = 1;
-            }
+            var numerator = (clientX - bbox.left) * this.params.pixelRatio;
+            var denominator = nominalWidth - 1;
         } else {
-            progress = ((clientX - bbox.left + this.wrapper.scrollLeft) / this.wrapper.scrollWidth) || 0;
+            var numerator = (clientX - bbox.left + this.wrapper.scrollLeft);
```

Comment:
"var" can only define once. How would you do it? I suspect an empty variable. I don't like that idea. Two extra superfluous lines of code.